tutorials/editor_basics/working_with_text.md

```diff
@@ -1,208 +1,163 @@
 ---
-title: 1. Working with text
+title: Working with text
 ---
 
-The following set of steps will show how to access a text selection and change it.
+The following set of steps will show how to access a text selection and replace the text with a string.
+The tutorial is presented in the following sections:
+* bullet list
+{:toc}
 
+## Introduction
+This tutorial relies heavily on creating and registering actions.
+The [Actions Tutorial](/tutorials/action_system.md) covers this topic in depth.
 
-## 1.1. Pre-requirements
+This tutorial uses examples from the [editor_basics](https://github.com/JetBrains/intellij-sdk-docs/tree/master/code_samples/editor_basics/) plugin code sample from the IntelliJ Platform SDK.
+It may be helpful to open that project in an IntelliJ Platform-based IDE, build the project, run it, select some text in the editor, and invoke the **Editor Replace Text** menu item on the editor context menu.
 
-### 1.1.1 Creating a new action
-
-In this example we access the editor from an action.
-To create an action we need to extend the
-[AnAction.java](upsource:///platform/editor-ui-api/src/com/intellij/openapi/actionSystem/AnAction.java)
-class.
-
-
-```java
-public class EditorIllustration extends AnAction {
-}
-```
-
-
-### 1.1.2. Registering an action
-
-To register the action we should add the corresponding tag to the `<actions>` section of the plugin configuration file
-[plugin.xml](https://github.com/JetBrains/intellij-sdk-docs/blob/master/code_samples/editor_basics/resources/META-INF/plugin.xml)
+![String replacement action](img/basics.png){:width="500px"}
 
+## Creating a New Menu Action
+In this example, we access the editor from an action.
+To review the fundamentals of creating and registering actions, refer to the [Actions Tutorial](/tutorials/action_system.md).
```

Comment:
Actions tutorial was just referenced in "introduction" above

Reply:
Sorry, sloppy.